validator/client/validator_attest.go

```diff
@@ -136,7 +146,23 @@ func (v *validator) SubmitAttestation(ctx context.Context, slot uint64, pubKey [
 		AggregationBits: aggregationBitfield,
 		Signature:       sig,
 	}
-
+	if v.protector != nil {
```

Comment:
I'd add the `SlasherProtection` check here for readability

Reply:
just thought that this protector can be implemented differently then by `SlasherProtection` wdyt?

Reply:
its probably better to emphasize that we need the flag that we need this to work. On an initial look it wouldn't be clear